llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2215,6 +2215,33 @@ template <> Value *SPIRVToLLVM::transValueWithOpcode<OpLoad>(SPIRVValue *const s
 
   SPIRVType *const spvLoadType = spvLoad->getSrc()->getType();
 
+  // We like to check if accesses to local arrays with dynamic indices are out of bounds.
+  bool shouldGenerateBoundaryCheck = false;
+  SPIRVToLLVM::ScratchBoundaryCheckInfo boundaryCheckInfo;
+  std::tie(shouldGenerateBoundaryCheck, boundaryCheckInfo) = prepareScratchBoundaryCheck(spvLoad);
+
+  if (shouldGenerateBoundaryCheck) {
+    getBuilder()->CreateCondBr(boundaryCheckInfo.finalCmpResult, boundaryCheckInfo.intermediateBlock,
+                               boundaryCheckInfo.finalBlock);
+
+    // Add the load instruction to the load block,
+    // followed by an unconditional jump.
```

Comment:
Can you add some simple ascii art showing how the transformation looks like?

Reply:
It's difficult to tell because there's no place in the PR that shows how the new IR looks like compared to the regular compilation. Maybe a diff would be more helpful than a diagram.

Reply:
I added some examples.